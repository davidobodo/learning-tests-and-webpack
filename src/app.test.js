import React from 'react';
import App from './app';
import { cleanup, fireEvent, render } from '@testing-library/react';


afterEach(cleanup);

// it('should take a snapshot', () => {
//     const { asFragment } = render(<App />);

//     expect(asFragment(<App />)).toMatchSnapshot;
// })

it('Full name should change input value', () => {
    const input = render(<App />).getByPlaceholderText('Obodo David');;
    fireEvent.change(input, { target: { value: 'rukky' } })
    expect(input.value).toBe('rukky')
})

it('Phone number should contain only letters', () => {
    const input = render(<App />).getByPlaceholderText('080XXXXXXXX');;
    fireEvent.change(input, { target: { value: '78ruk9ky' } })
    expect(input.value).toBe('789')
})

it('should display a button', () => {
    // const button = render(<App />).ge
})


















// test('render correctly inputs', () => {
//     let component = renderer.create(<App />).toJSON();
//     expect(component).toMatchSnapshot();
// })


//enzyme
// it('should change state after change in input value', () => {
//     let component;
//     let newValue = 'hello';
//     const wrapper = mount(<Auth />);

//     expect(wrapper.find('input[name="fullName"]').prop('value')).toEqual('david');
//     wrapper.find('input[name="fullName"]').simulate('change', { target: { name: 'fullName', value: newValue } });
//     expect(wrapper.find('input[name="fullName"]').prop('value')).toEqual(newValue)
// })
