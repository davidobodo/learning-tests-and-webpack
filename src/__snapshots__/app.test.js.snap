// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-AxhCb cNhBOm"
  >
    <form
      class="sc-AxirZ dLptuK"
    >
      <div
        class="sc-AxjAm kUeQQo"
      >
        <input
          class="input-field"
          name="fullName"
          placeholder="Obodo David"
          type="text"
          value=""
        />
        <label
          class="input-label"
        >
          Full Name
        </label>
      </div>
      <div
        class="sc-AxjAm kUeQQo"
      >
        <input
          class="input-field"
          name="email"
          placeholder="obododavid@gmail.com"
          type="email"
          value=""
        />
        <label
          class="input-label"
        >
          Email
        </label>
      </div>
      <div
        class="sc-AxjAm kUeQQo"
      >
        <input
          class="input-field"
          name="phone"
          placeholder="080XXXXXXXX"
          type="tel"
          value=""
        />
        <label
          class="input-label"
        >
          Phone Number
        </label>
      </div>
      <div
        class="sc-AxjAm kUeQQo"
      >
        <input
          class="input-field"
          name="password"
          placeholder="Password"
          type="password"
          value=""
        />
        <label
          class="input-label"
        >
          Password
        </label>
      </div>
      <div
        class="sc-AxjAm kUeQQo"
      >
        <input
          class="input-field"
          name="confirmPassword"
          placeholder="Confirm Password"
          type="password"
          value=""
        />
        <label
          class="input-label"
        >
          Confirm Password
        </label>
      </div>
      <div
        class="sc-AxjAm kUeQQo"
      >
        <input
          class="input-field"
          name="cardNumber"
          placeholder="XXXX XXXX XXXX XXXX"
          type="text"
          value=""
        />
        <label
          class="input-label"
        >
          Card Number
        </label>
      </div>
      <div
        class="sc-AxjAm kUeQQo"
      >
        <input
          class="input-field"
          name="expiryDate"
          placeholder="mm/yy"
          type="text"
          value=""
        />
        <label
          class="input-label"
        >
          Expiry Date
        </label>
      </div>
      <div
        class="sc-AxjAm kUeQQo"
      >
        <input
          class="input-field"
          name="pin"
          placeholder="****"
          type="password"
          value=""
        />
        <label
          class="input-label"
        >
          Pin
        </label>
      </div>
      <button
        class="sc-AxiKw eutMqT"
        data-testid="submit-button"
        disabled=""
      >
        Submit
      </button>
    </form>
  </div>
</DocumentFragment>
`;
